### ISSUE TITLE
[Feature]: Reduce code duplication between memory GCD and IO GCD

### ISSUE THREAD

**Issue by berlin-with0ut-return** [OpenDevicePartnership/patina#150]
### Feature Overview

There is heavy code duplication between the memory GCD and IO GCD (for example, compare `add_memory_space` and `add_io_space`). Evaluate if there is a better way to reuse code between these two structs. 

https://dev.azure.com/microsoft/MsUEFI/_git/DxeRust/pullrequest/10776793?_a=files has a POC of combining the two GCDs.

### Urgency

Low

### Are you going to implement the feature request?

None

### Do you need maintainer feedback?

None

**Comment by makubacki** [comment]
@Javagedes, should we close this?

**Comment by Javagedes** [comment]
@makubacki I don't think it should be closed, but realistically it is a very low priority as it would just help with code de-duplication (not even code reduction in the final binary) for easier maintainability. Due to that, if you want to close this just to get it cleaned up, you can.

**Comment by makubacki** [comment]
I'm trying to reduce our issue count to items we can take action on. Since this has been open since November, with no functional impact, and it's still unassigned, I'm closing it for now.
